food-truck-api/src/test/java/food/truck/api/endpoint/TruckEndpointTest.java

```diff
@@ -1,69 +1,183 @@
+        var req = delete("/truck/delete/{id}", data.testTruckA.getId())
+                .with(user(data.ownerB));
+        mockMvc.perform(req)
+                .andExpect(status().is4xxClientError());
+        assertEquals(1, truckService.findTruck(data.testTruckA.getName()).size());
+    }
+
+    @Test
+    public void getOwnTruck() throws Exception {
+        var req = get("/truck/owner")
+                .with(user(data.ownerA));
+        var strResp = mockMvc.perform(req)
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString();
+        var trucks = fromJsonList(strResp, Truck.class);
+        assertThat(trucks, contains(data.testTruckA));
+    }
+
+    @Test
+    public void getOwnTruckStandardUserFail() throws Exception {
+        var req = get("/truck/owner")
+                .with(user(data.standardUser));
+        mockMvc.perform(req)
+                .andExpect(status().is4xxClientError());
+    }
+
+    @Test
+    public void updateTruck() throws Exception {
+        var req = put("/truck/update")
+                .content(asJson(new TruckEndpoint.UpdateTruckParams(data.testTruckA.getId(), "truck1337", "a cool truck",
+                        3L, null, null, null, null)))
+                .contentType("application/json")
+                .with(user(data.ownerA));
+        String resp = mockMvc.perform(req)
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString();
+        Truck t = fromJson(resp, Truck.class);
+        assertEquals("truck1337", t.getName());
+        assertEquals("a cool truck", t.getDescription());
+        assertEquals(3L, t.getPriceRating());
+        assertNull(t.getFoodCategory());
+        assertNull(t.getMenu());
+        assertNull(t.getTextMenu());
+        assertNull(t.getSchedule());
+
+        var truck = truckService.findTruck("truck1337").get(0);
+        assertEquals(truck, t);
+    }
+
+    @Test
+    public void updateOtherOwnersTruckFail() throws Exception {
+        var req = put("/truck/update")
+                .content(asJson(new TruckEndpoint.UpdateTruckParams(data.testTruckA.getId(), "truck1337", "a cool truck",
+                        3L, null, null, null, null)))
+                .contentType("application/json")
+                .with(user(data.ownerB));
+        mockMvc.perform(req)
+                .andExpect(status().is4xxClientError());
+    }
+
+    @Test
+    @Transactional
+    public void getRoutes() throws Exception {
+        var req = get("/truck/{id}/routes", data.testTruckA.getId());
+        String resp = mockMvc.perform(req)
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString();
+        var list = fromJsonList(resp, Route.class);
+        assertThat(list, contains(data.testRouteA));
+    }
+
+    @Test
+    public void getRoutesBadId() throws Exception {
+        var req = get("/truck/{id}/routes", 83579233);
```

Comment:
I'm guessing this magic number is GUARANTEED to never be valid? Why not use -1 or a similar definitionally impossible ID?